plugins/inputs/oomkiller/oomkiller.go

```diff
@@ -0,0 +1,71 @@
+package oomkiller
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/tail"
+)
+
+type Oomkiller struct {
+	Logfile     string `toml:"logfile"`
+}
+
+var sampleConfig = `
+  ## Logfile where oom killer is reflected
+  logfile = "/var/log/messages"
+`
+
+// SampleConfig returns a sample configuration block
+func (a *Oomkiller) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the Mesos plugin
+func (a *Oomkiller) Description() string {
+	return "Telegraf plugin for gathering metrics from kernel Oomkiller"
+}
+
+func (a *Oomkiller) SetDefaults() {
+	if a.Logfile == "" {
+		log.Println("I! [oomkiller] Missing logfile value, setting default value (/var/log/messages)")
+		a.Logfile = "/var/log/messages"
+	}
```

Comment:
Check somewhere that the file exists and is readable by the plugin

Reply:
This is handled by the go tail library at the following lines: https://github.com/hpcloud/tail/blob/faf842bde7ed83bbc3c65a2c454fae39bc29a95f/tail.go#L192